tensorflow/core/distributed_runtime/master.cc

```diff
@@ -273,6 +282,17 @@ class DeviceFinder {
           }
         }
       }
+      if (env_->env->NowMicros() - start_time >
+          wait_for_worker_timeout_micros_) {
+        std::string unseen_workers;
```

Comment:
This looks highly inefficient, maybe use vector of `absl::string_view` and then concat with `absl::StrJoin`?